lib/rspec/rails/matchers/action_mailbox.rb

```diff
@@ -0,0 +1,64 @@
+module RSpec
+  module Rails
+    module Matchers
+      # Namespace for various implementations of ActionMailbox features
+      #
+      # @api private
+      module ActionMailbox
+        # @private
+        class Base < RSpec::Rails::Matchers::BaseMatcher
+          private
+
+          def create_inbound_email(message)
+            RSpec::Rails::MailboxExampleGroup.create_inbound_email(message)
+          end
+        end
+
+        # @private
+        class ReceiveInboundEmail < Base
+          def initialize(message)
+            super()
+
+            @inbound_email = create_inbound_email(message)
+          end
+
+          def matches?(mailbox)
+            @mailbox  = mailbox
+            @receiver = ApplicationMailbox.router.send(:match_to_mailbox, inbound_email)
```

Comment:
`actionmailbox`'s public API does not expose a way to check which mailbox an email routes to that doesn't fully send the email to the mailbox for processing. `expect(mailbox).to receive(:receive)` also seems fraught because of the method name there.

I'll open up a PR upstream to expose something like this in the public API. It seems like a generally useful thing for testing, and I hope won't be too hard to land. If it does, I'll update the implementation here.

Reply:
PR for that - https://github.com/rails/rails/pull/36181